### PR TITLE
[MRG+1] Change default S3 ACL to "private" and allow customization via settings

### PR DESCRIPTION
**Pull request by lopuhin**
See #1778 - this will make files uploaded via S3 private by default. If no settings customization is required, I can leave only the ``S3FilesStore.POLICY`` change.

### CODE REVIEW

**Comment by redapple** [comment]
+1 for the setting
I'd put this in 1.1 with a release notes mention

File: scrapy/pipelines/files.py [hunk 1]
```diff
@@ -232,6 +233,7 @@ def from_settings(cls, settings):
         s3store = cls.STORE_SCHEMES['s3']
         s3store.AWS_ACCESS_KEY_ID = settings['AWS_ACCESS_KEY_ID']
         s3store.AWS_SECRET_ACCESS_KEY = settings['AWS_SECRET_ACCESS_KEY']
+        s3store.POLICY = settings['S3_STORE_ACL']
```
Comment (dangra) [inline scrapy/pipelines/files.py]: IMO the setting name should be `FILES_STORE_S3_ACL`

Comment (dangra) [inline scrapy/pipelines/files.py]: The PR proposes to change policy of objects uploaded by feed exporters, this pipeline is not used for that.

You should look at scrapy/extensions/feedexporters.py for S3FeedStorage instead.

Comment (lopuhin) [inline scrapy/pipelines/files.py]: Thanks for the heads up, @dangra - I was updating docs in the wrong place, should be fixed in 6137dd9. Files uploaded by feed exporters should already have ``private`` permissions - in this PR I want to make ``S3FilesStore`` use private ACL.

**Comment by dangra** [comment]
I  don't mind accepting this change, but I want to share some historic references on why we defaulted to "public-read".

FilesPIpeline has its origin on a factorization of ImagesPipeline. We developed ImagesPipeline for a client that needed to download products from e-shops and then show them aggregated in a nicer UI. Uploading the images as `public-read` was convenient as this images were linked directly from client website and served from S3 bucket.

I still think this use case is common for ImagesPipeline, and even if this change is flagged as backwards-imcompatible our Users (new and old) will find an unpleasent gotcha when their images can't be used the way they expected by default.

So, the question is do we really want to change images default policy? if so, let's make it clear in images pipeline documentation.

**Comment by redapple** [comment]
@dangra , I would go with private by default, proper warning in release notes, and taking advantage of RC2 so users can tell us if it's a good move or not (sure, some users would discover only when 1.1 is officially baked)

@kmike , @eliasdorneles , @curita , @jdemaeyer , any comments ?

**Comment by jdemaeyer** [comment]
+1 for this change, I would say *explicit is better than implicit* also (and particularly) applies to making something publicly accessible. Maybe we should seize the chance to [actually document](https://github.com/scrapy/scrapy/blame/master/docs/topics/media-pipeline.rst#L144) this feature?

**Comment by eliasdorneles** [comment]
+1 from me, I also agree that releasing this in RC2 with proper documentation is a good way to test waters.

**Comment by redapple** [comment]
Needs backporting to 1.1

**Pull request merged**